docs/shallow_water_equation.md

```diff
@@ -0,0 +1,194 @@
+\end{equation}
+
+where
+
+\begin{align}
+  \mathbf{R}
+  =
+  \begin{bmatrix}
+  1 & 0 & 1  \\[.5em]
+  \hat{u} - \hat{c}\cos\phi & -\sin\phi & \hat{u} + \hat{c}\cos\phi \\[.5em]
+  \hat{v} - \hat{c}\sin\phi & \cos\phi & \hat{v} + \hat{c}\sin\phi
+  \end{bmatrix}
+\end{align}
+
+\begin{align}
+  \mathbf{\Delta\hat{V}}
+  =
+  \begin{bmatrix}
+  \frac{1}{2} \left( \Delta h - \frac{\hat{h}\Delta u_\perp}{\hat{a}} \right) \\[.5em]
+  \hat{h}u_\parallel \\[.5em]
+  \frac{1}{2} \left( \Delta h + \frac{\hat{h}\Delta u_\perp}{\hat{a}} \right)
+  \end{bmatrix}
+\end{align}
+
+\begin{eqnarray}
+  \hat{h} & = & \sqrt{h_i h_j} \\
+  \hat{u} & = & \frac{ \sqrt{h_i} u_i + \sqrt{h_j} u_j}{ \sqrt{h_i} + \sqrt{h_j}} \\
+  \hat{v} & = & \frac{ \sqrt{h_i} v_i + \sqrt{h_j} v_j}{ \sqrt{h_i} + \sqrt{h_j}} \\
+  \hat{a} & = & \sqrt{\frac{g}{2} \left( h_i + h_j \right)}
+\end{eqnarray}
+
+\begin{align}
+  |\mathbf{\hat{\Lambda}}|
+  =
+  \begin{bmatrix}
+  | \hat{u}_\perp - \hat{a} |^* & 0 & 0  \\[.5em]
+  0                                     & |\hat{u}_\perp| & 0 \\[.5em]  
+  0                                     &                         & | \hat{u}_\perp + \hat{a} |^* 
+  \end{bmatrix}
+\end{align}
+
+The asterisks denote that the eigenvalues 
+$\hat{\lambda}_1 (= \hat{u}_\perp - \hat{a} )$ and
+$\hat{\lambda}_3 (= \hat{u}_\perp + \hat{a})$ 
+are adjusted since Roe's method does not provide correct flux for critical flow.
+
+\begin{eqnarray}
+  |\hat{\lambda}|_1 &=& \frac{\hat{\lambda}^2_1}{\Delta \lambda} + \frac{\Delta \lambda}{4} \mbox{\hspace{0.5cm}if $-\Delta \lambda/2 < \hat{\lambda}_1 < \Delta \lambda/2$} \\
+  |\hat{\lambda}|_3 &=& \frac{\hat{\lambda}^2_2}{\Delta \lambda} + \frac{\Delta \lambda}{4} \mbox{\hspace{0.5cm}if $-\Delta \lambda/2 < \hat{\lambda}_3 < \Delta \lambda/2$}
+\end{eqnarray}
+ 
+### Source term
+
+The source term for the x-momentum equation
+
+\begin{eqnarray}
+\int_{d\Omega} \mathbf{S} d\Omega &=& \int_{\Omega} \left( -gh\frac{\partial z}{\partial x} - C_D u \sqrt{u^2 + v^2} \right) d\Omega \nonumber \\
+&=& \int_{d\Omega}  -gh\frac{\partial z}{\partial x} d\Omega - \int_{\Omega} C_D u \sqrt{u^2 + v^2}  d\Omega 
+\end{eqnarray}
+
+#### Bed slope elevation term
+
+\begin{eqnarray}
+\int_{d\Omega}  -gh\frac{\partial z}{\partial x} d\Omega &\approx& -gh\overline{\frac{\partial z}{\partial x}} \Omega \nonumber \\[0.6em]
+&=& \frac{(y_2 - y_0)(z_1 - z_0) - (y_1 - y_0)(z_2 - z_0)}{(y_2 - y_0)(x_1 - x_0) - (y_1 - y_0)(x_2 - x_0)} \Omega
+\end{eqnarray}
+
+#### Roughness term
+
+\begin{eqnarray}
+\int_{\Omega} C_D u \sqrt{u^2 + v^2}  d\Omega  &\approx& C_D u \sqrt{u^2 + v^2} \Omega
+\end{eqnarray}
+
+## PETSc TS implementation
+
+The PETSc TS notation\footnote{https://petsc.org/release/docs/manual/ts/} for solving time-dependent is given as:
+
+\begin{equation}
+\label{eqn:petsc_ts}
+\mathtt{ F(t,u,{u}_t)=G(t,u)}
+\end{equation}
+
+By comparing \ref{eqn:swe} and \ref{eqn:petsc_ts},
```

Comment:
\ref{eqn:swe} and \ref{eqn:petsc_ts} are not interpreted in latex.

Reply:
MkDocs (and probably Sphinx too) use [MathJax](https://docs.mathjax.org/en/latest/) for rendering mathematical formulae. According to [this](https://docs.mathjax.org/en/latest/input/tex/extensions/textmacros.html?highlight=eqref#equation-numbers), `\ref` and `\eqref` are supported in the latest spec, but I'm not sure we're using that.

Reply:
done